opm/material/common/PolynomialUtils.hpp

```diff
@@ -309,6 +309,105 @@ unsigned invertCubicPolynomial(SolContainer* sol,
 
     return 3;
 }
+
+/*!
+ * \ingroup Math
+ * \brief Invert a cubic polynomial analytically
```

Comment:
Just out of curiosity, why do we need an exact cubic solver instead of something resembling a Newton loop or eigenvalues of the companion matrix for that matter?

Reply:
Newton requires initial guesses and tolerances while the explicit roots are straightforward to compute and crucially allows AD to pass straight through which is more tricky if we call a algorithm.

Reply:
> crucially allows AD to pass straight through which is more tricky if we call an algorithm.

Right, I didn't think of that aspect.  I suppose another alternative would be [Laguerre's method](https://en.wikipedia.org/wiki/Laguerre%27s_method) with deflation, but since we know we're solving a cubic polynomial using the exact formulae is a reasonable strategy despite the messy calculations.